### PR TITLE
Frontend portfolio loading

### PR DESCRIPTION
**Pull request by thisconnect**
- should flicker less during loading, due to fixed widths
- use animated text spinner

tested with:
- just `make servewallet`
- `rm -rf appfolder.dev && cp -r appfolder.dev.brokentxdate appfolder.dev && make servewallet` Note: required new pairing, best unplug BitBox02 first
- `rm -rf appfolder.dev/cache/exchangerates && make servewallet`

### CODE REVIEW

File: frontends/web/src/components/spinner/ascii.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { h } from 'preact';
+import * as style from './ascii.css';
+
+export default function AsciSpinner() {
```
Comment (benma) [inline frontends/web/src/components/spinner/ascii.tsx:20]: Ascii?

File: frontends/web/src/routes/account/summary/accountssummary.tsx [hunk 2]
```diff
@@ -205,35 +205,43 @@ class AccountsSummary extends Component<Props, State> {
...
-                                isUpToDate={data.chartIsUpToDate} />}
+                            {data ? (
+                                <Chart
+                                    dataDaily={data.chartDataMissing ? undefined : data.chartDataDaily}
+                                    dataHourly={data.chartDataMissing ? undefined : data.chartDataHourly}
+                                    fiatUnit={data.chartFiat}
+                                    total={data.chartTotal}
+                                    isUpToDate={!data.chartDataMissing || data.chartIsUpToDate} />
+                            ) : (
+                                <p style="text-align: center;">loading chart…</p>
```
Comment (benma) [inline frontends/web/src/routes/account/summary/accountssummary.tsx:236]: Could you maybe just reword the 'gathering information' text and show that here instead? so we don't add too many i18n entries.

File: frontends/web/src/routes/account/summary/chart.tsx [hunk 3]
```diff
@@ -161,7 +163,9 @@ class Chart extends Component<Props, State> {
                     this.chart.subscribeCrosshairMove(this.handleCrosshair as MouseEventHandler);
                 }
             }
-            this.chart.timeScale().fitContent();
+            if (this.chart) {
```
Comment (benma) [inline frontends/web/src/routes/account/summary/chart.tsx:166]: isn't this always true?

File: frontends/web/src/components/headerssync/headerssync.tsx [hunk 4]
```diff
@@ -80,3 +92,10 @@ export default class HeadersSync extends Component {
         );
     }
 }
+
+const subscribeHOC = subscribe<SubscribedHeadersSyncProps, HeadersSyncProps & TranslateProps>(({ coinCode }) => ({
+    status: `coins/${coinCode}/headers/status`,
+}), false, true)(HeadersSync);
```
Comment (benma) [inline frontends/web/src/components/headerssync/headerssync.tsx:98]: why `..., false, true`? Before it seemed to be the opposite.

File: frontends/web/src/components/headerssync/headerssync.tsx [hunk 5]
```diff
@@ -1,5 +1,6 @@
+
```
Comment (benma) [inline frontends/web/src/components/headerssync/headerssync.tsx:1]: This change seems completely unrelated to the chart, so it could have been a separate PR :)

**Review by benma — APPROVED** [review]

**Comment by benma** [comment]
When rebasing, use `formatCurrency(total, fiatUnit)` in the conflicting line.

File: frontends/web/src/components/headerssync/headerssync.tsx (hunk 5 above)
Comment (thisconnect) [reply frontends/web/src/components/headerssync/headerssync.tsx:1]: Refactored, because in the version where the headerssync was displayed in the sumamrytable.

What is left is  we share the same spinner. Can we keep it in the same PR but different commit?

File: frontends/web/src/routes/account/summary/accountssummary.tsx (hunk 2 above)
Comment (thisconnect) [reply frontends/web/src/routes/account/summary/accountssummary.tsx]: done

File: frontends/web/src/routes/account/summary/chart.tsx (hunk 3 above)
Comment (thisconnect) [reply frontends/web/src/routes/account/summary/chart.tsx]: done

File: frontends/web/src/components/headerssync/headerssync.tsx (hunk 4 above)
Comment (thisconnect) [reply frontends/web/src/components/headerssync/headerssync.tsx]: will open a separate PR after this is merged

File: frontends/web/src/components/headerssync/headerssync.tsx (hunk 5 above)
Comment (thisconnect) [reply frontends/web/src/components/headerssync/headerssync.tsx]: will open a separate PR

File: frontends/web/src/routes/account/summary/accountssummary.tsx [hunk 6]
```diff
@@ -205,35 +205,45 @@ class AccountsSummary extends Component<Props, State> {
...
                     </Header>
                     <div className="innerContainer scrollableContainer">
                         <div className="content padded">
-                            {data && <Chart
-                                dataDaily={data.chartDataMissing ? undefined : data.chartDataDaily}
-                                dataHourly={data.chartDataMissing ? undefined : data.chartDataHourly}
-                                fiatUnit={data.chartFiat}
-                                total={data.chartTotal}
-                                isUpToDate={data.chartIsUpToDate} />}
+                            {(data && data.chartDataDaily && data.chartDataDaily.length) ? (
```
Comment (benma) [inline frontends/web/src/routes/account/summary/accountssummary.tsx:228]: i think this is not correct. there could be legitimately zero data points and the chart is still fully loaded (e.g. when there are no transactions).

before was correct.

**Review by benma — CHANGES_REQUESTED** [review]

File: frontends/web/src/routes/account/summary/accountssummary.tsx [hunk 7]
```diff
@@ -205,35 +205,45 @@ class AccountsSummary extends Component<Props, State> {
...
+                                        <line x1="12" y1="15" x2="12" y2="3"></line>
+                                    </svg>
+                                </a>
+                            )
+                        )}
                     </Header>
                     <div className="innerContainer scrollableContainer">
                         <div className="content padded">
-                            {data && <Chart
-                                dataDaily={data.chartDataMissing ? undefined : data.chartDataDaily}
```
Comment (benma) [inline frontends/web/src/routes/account/summary/accountssummary.tsx:230]: why is this gone? the two fields are only valid (defined) if chartDataMissing is false, that's why it was like this. the backend might deliver garbage in this case

File: frontends/web/src/components/headerssync/headerssync.tsx (hunk 4 above)
Comment (benma) [reply frontends/web/src/components/headerssync/headerssync.tsx]: why?

**Review by benma — APPROVED** [review]

File: frontends/web/src/routes/account/summary/accountssummary.tsx (hunk 7 above)
Comment (thisconnect) [reply frontends/web/src/routes/account/summary/accountssummary.tsx:230]: fixed
